src/main/java/ch/njol/skript/bukkitutil/SoundUtils.java

```diff
@@ -0,0 +1,69 @@
+package ch.njol.skript.bukkitutil;
+
+import org.bukkit.NamespacedKey;
+import org.bukkit.Sound;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Locale;
+
+public class SoundUtils {
+
+	private static final boolean SOUND_IS_INTERFACE;
+	private static final Method SOUND_GET_KEY;
+
+	static {
+		try {
+			Class<?> SOUND_CLASS = Class.forName("org.bukkit.Sound");
+			SOUND_IS_INTERFACE = SOUND_CLASS.isInterface();
+			SOUND_GET_KEY = SOUND_CLASS.getDeclaredMethod("getKey");
+		} catch (ClassNotFoundException | NoSuchMethodException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	/**
+	 * Gets the key of a sound, given its enum-name-style name.
+	 * @param soundString The enum name to use to find the sound.
+	 * @return The key of the sound.
+	 */
+	public static @Nullable NamespacedKey getKey(String soundString) {
+		soundString = soundString.toUpperCase(Locale.ENGLISH);
+		// Sound.class is an Interface (rather than an enum) as of MC 1.21.3
+		if (SOUND_IS_INTERFACE) {
+			try {
+				//noinspection deprecation
+				return Sound.valueOf(soundString).getKey();
+			} catch (IllegalArgumentException ignore) {
+			}
+		} else {
+			try {
+				//noinspection unchecked,rawtypes
+				Enum soundEnum = Enum.valueOf((Class) Sound.class, soundString);
+				return ((NamespacedKey) SOUND_GET_KEY.invoke(soundEnum));
```

Comment:
I could be wrong, but as per my last PR, I don't think the reflection is necessary.
```suggestion
				return return ((Keyed) soundEnum).getKey();
```
When Sound.class was an enum, it still implemented Keyed

Reply:
Oh i mistook that for the Adventure API Keyed
Didn't realize there was a bukkit one too.